File: 프로그래머스/lv2/42578. 위장/위장.py
def solution(clothes):
    answer = 1
    cloth = {}
    for i in clothes:
        if i[1] in cloth:
            cloth[i[1]] +=[i[0]]
        else:
            cloth[i[1]] = [i[0]]
    
    # 경우의 수 구하기            
    for value in cloth.values():
        answer *= len(value) + 1
    
    # 아무것도 입지 않은 경우 하나 제외
    return answer-1

File: 프로그래머스/lv2/42578. 위장/위장.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> map = new HashMap<>();
        
        for(int i=0; i<clothes.length; i++){
            String[] clothe = clothes[i];
            if(!map.containsKey(clothe[1])){
                map.put(clothe[1],1);
            }
            else{
                map.put(clothe[1],map.get(clothe[1])+1);
            }
        }
        // System.out.println(map);
        for(String s : map.keySet()){
            answer *=map.get(s)+1;
        }
        return answer-1;
    }
}